contrib/coredns/policy/policy.go

```diff
@@ -51,9 +55,16 @@ type PolicyMiddleware struct {
 	ErrorFunc func(dns.ResponseWriter, *dns.Msg, int) // failover error handler
 }
 
+type Attribute struct {
+	Id    string
+	Type  string
+	Value string
+}
+
 type Response struct {
-	Permit   bool   `pdp:"Effect"`
-	Redirect net.IP `pdp:"redirect_to"`
+	Permit     bool   `pdp:"Effect"`
+	Redirect   net.IP `pdp:"redirect_to"`
+	Obligation []*Attribute
```

Comment:
Yes. It should look like @johnbelamaric wrote above. You don't need Obligation array just this PolicyId field